### PR TITLE
Replace travis CI with GH actions

### LINKED ISSUE #223: Travis CI not running on master branch
Last run seems to be 3 years ago https://travis-ci.org/github/graalvm/mx/builds

Why is that? Could someone please re-enable it? 

PS: I am volunteering to port the existing configuration to github-actions.

### PR DESCRIPTION
**Pull request by zakkak**
Closes #223

### CODE REVIEW

File: .github/workflows/gate.yml [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+      - run: pylint --version
+      - name: mx gate
+        run: |
+          export ECLIPSE_TAR=${{ github.workspace }}/../eclipse.tar.gz
+          wget https://archive.eclipse.org/eclipse/downloads/drops4/R-4.5.2-201602121500/eclipse-SDK-4.5.2-linux-gtk-x86_64.tar.gz -O $ECLIPSE_TAR
+          tar -C ${{ github.workspace }}/.. -xf $ECLIPSE_TAR
+          export ECLIPSE_EXE=${{ github.workspace }}/../eclipse/eclipse
+          export JDT=${{ github.workspace }}/ecj.jar
+          wget https://www.eclipse.org/downloads/download.php?file=/eclipse/downloads/drops4/R-4.5.2-201602121500/ecj-4.5.2.jar -O $JDT
+          ./mx --strict-compliance gate --strict-mode
```
Comment (fniephaus) [inline .github/workflows/gate.yml:30]: Should this step be splitted into two steps, e.g. "Set up dependencies" and "Run mx gate"? That way it's easier to see if the gate failed or if the worker just failed to download a dependency.

**Comment by dougxc** [comment]
Why is the GitHub action not being run in this PR?

File: .github/workflows/gate.yml (hunk 1 above)
Comment (zakkak) [reply .github/workflows/gate.yml:30]: Yes, that would be better. Thanks for the comment

**Comment by zakkak** [comment]
> Why is the GitHub action not being run in this PR?

I suspect because the PR was opened before the actions were enabled in the repository.

**Comment by zakkak** [comment]
Force push still didn't trigger it.
You can see it running here https://github.com/zakkak/mx/runs/1002669483 or I can close this PR and open a new one to see if it will trigger this way.

**Comment by fniephaus** [comment]
The action will only trigger in this repo if it's part of the repository...so after it's merged.

**Review by fniephaus — APPROVED** [review]
LGTM

**Comment by dougxc** [comment]
> The action will only trigger in this repo if it's part of the repository...so after it's merged.

So how does one ensure GitHub action updates don't break the gate?

**Comment by zakkak** [comment]
> > The action will only trigger in this repo if it's part of the repository...so after it's merged.
> 
> So how does one ensure GitHub action updates don't break the gate?

After the merge, it will be part of the repo and it will then be triggered on pull requests (for any change in the repo due to https://github.com/graalvm/mx/pull/224/files#diff-6290e0dc7d30be4ffb4222544eb22290R1)

**Comment by dougxc** [comment]
Ok. Looks ready to merge then.

**Comment by dougxc** [comment]
Once we have clearance to use the third party actions in this PR (e.g., `- uses: actions/setup-java@v1`), I'll merge it. (/cc @RoxaBee )

**Comment by zakkak** [comment]
@dougxc @RoxaBee any updates on this?

**Comment by dougxc** [comment]
Sorry for not updating, but it was merged in https://github.com/graalvm/mx/commit/ebcfdd01f65d7961aae063f584cb0e4d849393cc. However, it seems to have broken the gate (e.g. https://github.com/graalvm/mx/runs/1113299642#step:8:38). Can you please look into that.

**Comment by zakkak** [comment]
Yup, please see #227 :)

**Pull request closed without merging**